### ISSUE TITLE
How to use Lumberjack as a framework for cocoapods

### ISSUE THREAD

**Issue by KeGao** [CocoaLumberjack/CocoaLumberjack#891]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](https://github.com/CocoaLumberjack/CocoaLumberjack#documentation) or [CocoaDocs](cocoadocs.org/docsets/CocoaLumberjack)
* [ ] I have searched for a similar issue in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | macOS Sierra
 Platform Version        | 10.12.5 (16F73)
 CocoaLumberjack Version | 3.x
 Integration Method      | cocoapods
 Xcode Version           | 8.3.3 (8E3004b)
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

I'm using CocoaLumberjack via CocoaPods,  When i write `#import <CocoaLumberjack.h>`  it's ok, But i write `@import CocoaLumberjack;` it's an error that `Module 'CocoaLumberjack' not found`, how to fix it

**Comment by vladdeSV** [comment]
I am experiencing similar issues with other libraries useing Lumberjack.

**Comment by RabbitMC** [comment]
Specify your Xcode version, your Platform and the version of it :) This would help

**Comment by vladdeSV** [comment]
**macOS Sierra** 10.12.5 (16F73)
**Xcode** 8.3.3 (8E3004b)

The project I am using has the following line in its [`.podspec`](https://github.com/SVGKit/SVGKit/blob/2.x/SVGKit.podspec)

```
    s.dependency 'CocoaLumberjack', '~> 3.x'
```

**Comment by vladdeSV** [comment]
*(forgot pods)*

**CocoaPods** 1.3.1

**Comment by RabbitMC** [comment]
Swift or Obj-C?

**Comment by vladdeSV** [comment]
Objective C

**Comment by KeGao** [comment]
@RabbitMC My Xcode version is 8.3.3  and CocoaLumberjack version is 3.2.1, I don't know how to use @import CocoaLumberjack;  If i write that ,Syntax compile issue Module 'CocoaLumberjack' not found,maybe header search paths can not found CocoaLumberjack.modulemap

**Comment by KeGao** [comment]
@vladdeSV Did you experienced this issue while using SvgKit?

**Comment by vladdeSV** [comment]
Yes.

SVGKit imports Lumberjack with `@import CocoaLumberjack;`

**Comment by KeGao** [comment]
@vladdeSV use cocoapods and ignore pods ??  everytime use pod install or update need Replace @import CocoaLumberjack; with #import <CocoaLumberjack.h>?

**Comment by vladdeSV** [comment]
This is off-topic, but I created my own fork of SVGKit where I use `#import <CocoaLumberjack.h>`

However, that is not related to the issue.

**Comment by KeGao** [comment]
I just want to know if we have the same problem, and I have an Imperfect solution that can be used with @import CocoaLumberjack;  Write the following code into your podfile, and run "pod install"
def generate_modulemap(name, path)
    f = File.new(File.join("#{path}/module.modulemap"), "w+")
    module_name = "#{name}"
    while(module_name["+"])
        module_name["+"] = "_"
    end
    f.puts("module #{module_name} {")
    f.puts("    umbrella header \"#{name}_umbrella.h\"")
    f.puts("    export *")
    f.puts("}")
end

def generate_umbrella(name, path)
    f = File.new(File.join("#{path}/#{name}_umbrella.h"), "w+")
    f.puts("#import <Foundation/Foundation.h>")
    Dir.foreach(path) do |filename|
        if filename != "." and filename != ".."
            f.puts("#import \"#{filename}\"")
        end
    end
end

post_install do |installer|
    require "fileutils"
    headers_path = "#{Dir::pwd}/Pods/Headers/Public/"
    
    installer.pods_project.targets.each do |target|
        target_header_path = "#{headers_path}#{target.product_name}"
        if File.exist?(target_header_path)
            filename = target.product_name
            if filename != "." and filename != ".."
                generate_umbrella(filename, target_header_path)
                generate_modulemap(filename, target_header_path)
            end
        end
    end
end

**Comment by vladdeSV** [comment]
Thank you, but again: I feel this is off the topic and not helping the issue at hand.

**Comment by bpoplauschi** [comment]
I think this might be related to `use_frameworks!`. Do you have this line in your `Podfile`? If this is added, then all pods including our own will be built as frameworks, so the `@import framework_name` should definitely work. In the other case, for static libraries, only the `#import <CocoaLumberjack.h>` will work.
`@import framework_name` only works with frameworks.

**Comment by sushichop** [comment]
@KeGao 
Thanks for reporting.

If you still have the issue, let us know.
I recommend you use `CocoaLumberjack 3.4.2`(the latest) now :)

Try to use Podfile like the following, then you will be able to import **not  `<CocoaLumberjack.h>` but `<CocoaLumberjack/CocoaLumberjack.h>`** .

```ruby
use_frameworks!

target 'YourSampleApp' do
  pod 'CocoaLumberjack', '3.4.2'
end
```

**Comment by sushichop** [comment]
I'm going to close this issue based on the last comments.
Please feel free to reopen it if you still encounters this problem using `CocoaLumberjack 3.4.2` and Swift 4.
